src/libraries/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/NativeMemoryTests.cs

```diff
@@ -0,0 +1,327 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public unsafe class NativeMemoryTests
+    {
+        [Fact]
+        public void AlignedAllocTest()
```

Comment:
Add test that verifies that the memory returned by `AlignedAlloc` is actually aligned?

Also, we should exercise with large alignments, like 16MB.

Reply:
Added theory coverage for powers of two from `1 to 16384` on both `AlignedAlloc` and `AlignedRealloc`